app/containers/ITunesContainer/index.js

```diff
@@ -4,30 +4,64 @@
  *
  */
 
-import React, { memo } from 'react';
+import React, { useState, useEffect, memo } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { injectIntl } from 'react-intl';
-import { debounce, isEmpty } from 'lodash';
+import { debounce, isEmpty, get } from 'lodash';
 import styled from 'styled-components';
 import { injectSaga } from 'redux-injectors';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
-import { Card, Input } from 'antd';
-import makeSelectITunesContainer from './selectors';
+import { Card, Input, Skeleton } from 'antd';
+import ITunesCard from '@app/components/ITunesCard';
+import { selectITunesContainer, selectITunesSearchQuery, selectITunesData, selectITunesError } from './selectors';
 import { iTunesContainerCreators } from './reducer';
 import saga from './saga';
 
 const { Search } = Input;
 
 const CustomCard = styled(Card)`
   && {
-    margin: 20px 0;
+    margin: 1.25rem 0;
     max-width: ${(props) => props.maxwidth};
   }
 `;
+const Container = styled.div`
+  && {
+    display: grid;
+    grid-template-columns: repeat(3, 1fr);
+    gap: 1em;
+    padding: 1em;
+  }
+`;
+
+export function ITunesContainer({
+  dispatchITunesSongs,
+  dispatchClearITunesSongs,
+  searchQuery,
+  iTunesError,
+  iTunesData,
+  intl,
+  maxwidth
+}) {
+  const [currentSongRef, setCurentSongRef] = useState(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(iTunesData, 'results', null) || iTunesError;
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [iTunesData]);
+
+  useEffect(() => {
+    if (searchQuery && !iTunesData?.results?.length) {
```

Comment:
Where are we storing the searchQuery at ?

Reply:
`searchQuery` is stored in the **store**, along with `data` and `error`